### PR TITLE
Current work on adding unit testing. Added some example component tests

### LINKED ISSUE #219: X939 - Introduce javascript unit testing
**User story**
As the Dev Team we would like to introduce javascript unit testing to the front end so that not everything becomes an integration test.

**Acceptance**
* Select and introduce a javascript unit testing framework
* Incorporate the unit testing into the CI build
* Update setup documentation with instructions on installing the dependencies.

### PR DESCRIPTION
**Pull request by BenTopping**
Closes #219 

Current work on adding unit testing. 

Changes proposed in this pull request:

* Added some example tests: components, machines, services
* Added config for jest and ts-jest
* Added CI for jest tests

**TODO**
* Add tests for lib/validation and lib/hooks?

### CODE REVIEW

**Review by seenanair — APPROVED** [review]
Excellent job! This is a great improvement to introduce unit-testing to stan cllent side 👍

File: src/mocks/handlers/findHandlers.ts [hunk 1]
```diff
@@ -45,7 +45,7 @@ export function buildFindResult(numRecords: number, maxRecords: number): FindRes
   const locations = [...Array(locationCount)].map(() =>
     locationFactory.build({
       fixedName: faker.address.countryCode(),
-      customName: _.sample([faker.address.streetName(), null])
+      customName: _.sample([faker.address.street(), null])
```
[the remark this answers is not in the document]
Comment (BenTopping) [reply src/mocks/handlers/findHandlers.ts:48]: had a console warning that streetName is becoming deprecated in future versions and that street is the closest alternative

File: package.json [hunk 2]
```diff
@@ -113,12 +114,16 @@
     "cypress-intellij-reporter": "^0.0.7",
     "eslint-config-prettier": "^8.5.0",
     "fishery": "^2.2.2",
+    "identity-obj-proxy": "^3.0.0",
+    "jest": "28.1.3",
+    "jest-environment-jsdom": "^29.1.2",
```
[the remark this answers is not in the document]
Comment (BenTopping) [reply package.json:119]: jest-dom provides us custom elemtn matchers for dom components (allows us to test components more easily). I believe jest-environment-jsdom provides a config for jest to mount the dom components in a certain way.

File: package.json [hunk 3]
```diff
@@ -8,8 +8,8 @@
     "@sanger/ui-styling": "^4.0.0",
     "@tailwindcss/aspect-ratio": "^0.4.0",
     "@tailwindcss/typography": "^0.5.4",
-    "@testing-library/jest-dom": "^5.16.4",
-    "@testing-library/react": "^13.3.0",
+    "@testing-library/jest-dom": "^5.16.5",
+    "@testing-library/react": "12.1.5",
```
[the remark this answers is not in the document]
Comment (BenTopping) [reply package.json:12]: Yeah I think there was a compatibility issue with one of the other package versions. Will have a look into this.

**Pull request merged**